ably-ios/ARTAuth.m

```diff
@@ -135,42 +135,61 @@ - (void)requestToken:(ARTAuthTokenParams *)tokenParams withOptions:(ARTAuthOptio
     ARTAuthOptions *mergedOptions = [self mergeOptions:authOptions];
     ARTAuthTokenParams *currentTokenParams = [self mergeParams:tokenParams];
 
-    if (!mergedOptions.key) {
-        callback(nil, [NSError errorWithDomain:ARTAblyErrorDomain code:ARTCodeErrorAPIKeyMissing
-                                      userInfo:@{ NSLocalizedDescriptionKey: NSLocalizedString(@"API Key is missing", nil) }]);
-    }
-
     if (mergedOptions.authUrl) {
         NSMutableURLRequest *request = [self buildRequest:mergedOptions withParams:currentTokenParams];
         
         [self.logger debug:__FILE__ line:__LINE__ message:@"using authUrl (%@ %@)", request.HTTPMethod, request.URL];
         
-        [_rest executeRequest:request withAuthOption:ARTAuthenticationUseBasic completion:^(NSHTTPURLResponse *response, NSData *data, NSError *error) {
+        [_rest executeRequest:request withAuthOption:ARTAuthenticationOff completion:^(NSHTTPURLResponse *response, NSData *data, NSError *error) {
```

Comment:
`ARTAuthenticationOff` means the host does not require authentication (no `Authorization` http header)